addons/website_rating/static/src/js/website_mail.js

```diff
@@ -0,0 +1,186 @@
+                _.each(messages, function(m){
+                    m['rating_value'] = self.round_to_half(m['rating_value']);
+                });
+            }
+            return messages;
+        },
+        // rating card
+
+        // rating star input
+        onchange_star_value: function(){
+            var val = this.get("star_value");
+            var index = Math.floor(val);
+            var decimal = val - index;
+            // reset the stars
+            this.$star_list.removeClass('fa-star fa-star-half-o').addClass('fa-star-o');
+
+            this.$('.stars').find("i:lt("+index+")").removeClass('fa-star-o fa-star-half-o').addClass('fa-star');
+            if(decimal){
+                this.$('.stars').find("i:eq("+(index)+")").removeClass('fa-star-o fa-star fa-star-half-o').addClass('fa-star-half-o');
+            }
+            this.$input.val(val);
+            this.$('.rate_text .label').text(this.labels[index]);
+        },
+        star_move_out: function(){
+            if(!this.user_click){
+                this.set("star_value", 0);
+            }
+            this.user_click = false;
+        },
+        star_move_on: function(e){
+            var index = this.$('.stars i').index(e.currentTarget);
+            this.$('.rate_text').show();
+            this.set("star_value", index+1);
+        },
+        star_click_on: function(e){
+            this.user_click = true;
+        },
+        star_mouseleave: function(e){
+            this.$('.rate_text').hide();
+        },
+        // star card domain
+        onclick_star_domain: function(e){
+            var $tr = this.$(e.currentTarget);
+            var num = $tr.data('star');
+            if($tr.css('opacity') == 1){
+                this.set('rating_value', num);
+                this.$('.o_website_rating_select').css({
+                    'opacity': 0.5,
+                });
+                this.$('.o_website_rating_select_text[data-star="'+num+'"]').css({
+                    'visibility': 'visible',
+                    'opacity': 1,
+                });
+                this.$('.o_website_rating_select[data-star="'+num+'"]').css({
+                    'opacity': 1,
+                });
+            }
+        },
+        onclick_star_domain_reset: function(e){
+            e.stopPropagation();
+            this.set('rating_value', false);
+            this.$('.o_website_rating_select_text').css('visibility', 'hidden');
+            this.$('.o_website_rating_select').css({
+                'opacity': 1,
+            });
+        },
+        onchange_rating_domain_value: function(){
+            var domain = [];
+            if(this.get('rating_value')){
+                domain = [['rating_value', '=', this.get('rating_value')]];
+            }
+            console.log('change to ',this.get('rating_value'));
+            this.change_current_page(1, domain);
+        },
+        // utils
+        round_to_half: function(value) {
```

Comment:
I am not a JS expert, but don't we have access to some method allowing to do it :p ?